src/integrationTest/codecatalyst/client.test.ts

```diff
@@ -0,0 +1,169 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import {
+    CodeCatalystClient,
+    CodeCatalystOrg,
+    CodeCatalystProject,
+    createClient as createCodeCatalystClient,
+    DevEnvironment,
+} from '../../shared/clients/codecatalystClient'
+import { DeleteDevEnvironmentResponse } from 'aws-sdk/clients/codecatalyst'
+import { prepareDevEnvConnection } from '../../codecatalyst/model'
+import { Auth, createBuilderIdConnection } from '../../credentials/auth'
+import { isValidCodeCatalystConnection } from '../../codecatalyst/auth'
+
+let spaceName: CodeCatalystOrg['name']
+let projectName: CodeCatalystProject['name']
+
+/**
+ * Key Information:
+ *
+ * This test can be run locally under the following conditions:
+ *
+ *   - You have previously configured BuilderId using the vscode extension.
+ *     This is because this test uses BuilderId/SSO information
+ *     that the extension would have naturally created in that process.
+ *
+ *   - A space already exists in your Code Catalyst account.
+ *     We cannot currently automate this due to space creation
+ *     not being available in the Code Catalyst API.
+ *
+ *     TODO: Create a new space instead of using an existing one,
+ *     if the api eventually allows us to do so.
+ *
+ *   - You do not have a Code Catalyst project with the same name
+ *     as {@link projectName}. This project may be modified when
+ *     this test is run.
+ *
+ * The test project cannot be torn down.
+ *
+ *   - The code catalyst api does not provide a way to delete
+ *     a Project. So the project will exist in the Space unless
+ *     deleted manually.
+ *
+ *     TODO: Delete the project when the test is done,
+ *     if the api eventually allows us to do so.
+ *
+ * This provides a best effort to test the api functionality.
+ *
+ *   - We cannot currently perform integ tests with UI components,
+ *     so the underlying non-UI components that would be used in
+ *     those processes are being tested.
+ */
+describe('Test how this codebase uses the Code Catalyst API', () => {
+    let client: CodeCatalystClient
+
+    before(async () => {
+        client = await createTestCodeCatalystClient()
+        spaceName = (await getCurrentUsersSpace()).name
+        projectName = (await tryCreateTestProject(spaceName)).name
+        await deleteExistingDevEnvs(projectName)
+    })
+
+    /**
+     * Creates a code catalyst client.
+     *
+     * This relies on BuilderId/SSO information already being configured.
+     * See the above "Key Information" comment for more info.
+     */
+    async function createTestCodeCatalystClient(): Promise<CodeCatalystClient> {
+        const conn = await createBuilderIdConnection(Auth.instance)
+        assert(isValidCodeCatalystConnection(conn))
```

Comment:
`createBuilderIdConnection` may cause the test to hang until the user logs into an account which we can't do in CI:
```suggestion
        const conn = (await Auth.instance.listConnections()).find(isValidCodeCatalystConnection)
        assert.ok(conn, 'No valid CodeCatalyst connection found.')
```